litex/soc/interconnect/csr.py

```diff
@@ -268,10 +268,17 @@ def read(self):
 
     def write(self, value):
         """Write method for simulation."""
-        yield self.storage.eq(value >> self.alignment_bits)
-        yield self.re.eq(1)
+        for sc in reversed(self.simple_csrs):
+            yield from sc.write(value)
+            yield
```

Comment:
This should probably be something like;
```python
    for sc in reversed(self.simple_csrs):
        yield from sc.write(value)
    yield
```

Unsure if I tested if that version worked?